device/index.js

```diff
@@ -490,6 +490,13 @@ function DeviceClient(options) {
                }
                var user_profile = options.profile || process.env.AWS_PROFILE || 'default';
                var creds = getCredentials(fs.readFileSync(filename, 'utf-8'));
+
+                if (!creds[user_profile]) {
+                    console.log("WS Access Key ID and AWS Secret Key not found in file:" + options.filename + " for AWS_PROFILE:" + process.env.AWS_PROFILE);
+                    throw new Error(exceptions.INVALID_CONNECT_OPTIONS);
+                }
```

Comment:
This explicit check isn't necessary, the code that loads the credentials file and gets data out is already wrapped in a try-catch that logs a decent message if something goes wrong:
`console.log('Failed to read credentials from ' + filename);`

then a few lines later, another useful message is logged and the INVALID_CONNECT_OPTIONS error is thrown:
```
// AWS Access Key ID and AWS Secret Key must be defined
if (isUndefined(awsAccessId) || (isUndefined(awsSecretKey))) {
   console.log('To connect via WebSocket/SigV4, AWS Access Key ID and AWS Secret Key must be passed either in options or as environment variables; see README.md');
   throw new Error(exceptions.INVALID_CONNECT_OPTIONS);
}
```

Maybe just improve the existing log message to include the user_profile it tried to load:
`console.log('Failed to read credentials for profile "' + user_profile + '" from ' + filename);`

You'll need to move the declaration of `var user_profile` outside the try-block, like `var filename`

Reply:
Thanks, it was helpful.
I have updated the PR.

Reply:
Up updated.